airflow/www/app.py

```diff
@@ -980,25 +984,27 @@ def action(self):
                     t.task_id
                     for t in task.get_flat_relatives(upstream=True)]
             TI = models.TaskInstance
+            dates = utils.date_range(start_date, end_date)
```

Comment:
Yeah, 1k? We'll have to see how long it takes to do 1k. Shouldn't be more than 15 - 30 seconds. With the cap on the cartesian product along with the fix bellow you could eventually get it through by repeating the same command N times.